operator/helm/helm.go

```diff
@@ -210,6 +202,45 @@ func (c *client) ListResources(rel *release.Release, opts ListResourcesOpts) (ku
 	return current, nil
 }
 
+// IsReady determines whether a Helm installation is ready or not.
+func (c *client) IsReady(rel *release.Release) (bool, error) {
+	nsName := namespacedName{
+		name:      rel.Name,
+		namespace: rel.Namespace,
+	}
+	ch, ok := c.readyReleases[nsName]
+	if ok {
+		select {
+		case err := <-ch:
+			close(ch)
+			delete(c.readyReleases, nsName)
```

Comment:
Doesn't this mean this will keep flopping between ready and not, on alternate calls?
* Get here the first time: delete the map entry, return `true`.
* Next call: map entry is missing, set up the wait, return `false`.
* Next call: go to 1.

Reply:
No, the `select` is non-blocking and select the default case if the `ch` channel doesn't have anything to read.
The `ch` channel will only have 1 `error` or `nil` **only** when the `Wait` finishes.

Reply:
No, I'm talking about when things actually finish.

* Initial run: start the wait, returning `false`.
* Runs in the middle: waiting, returning `false`.
* Wait finishes: delete entry, return `true`.
* Next time it gets called again: entry is now missing, returns `false` again (setting up for the next wait)

Reply:
Correct. I was going to rely on the caller to determine what it should do on the first `false` return. This is not very robust and not really what the `IsReady` describes as its behaviour. I'm introducing a timeout for when `Wait` spawns. 0f58969